src/tools/rbd_mirror/Replayer.cc

```diff
@@ -228,26 +83,22 @@ class MirrorStatusWatchCtx {
 };
 
 Replayer::Replayer(Threads *threads, std::shared_ptr<ImageDeleter> image_deleter,
-                   RadosRef local_cluster, const peer_t &peer,
-                   const std::vector<const char*> &args) :
+                   RadosRef local_cluster, int64_t local_pool_id,
+                   const peer_t &peer, const std::vector<const char*> &args) :
   m_threads(threads),
   m_image_deleter(image_deleter),
   m_lock(stringify("rbd::mirror::Replayer ") + stringify(peer)),
   m_peer(peer),
   m_args(args),
   m_local(local_cluster),
   m_remote(new librados::Rados),
-  m_asok_hook(nullptr),
+  m_local_pool_id(local_pool_id),
   m_replayer_thread(this)
 {
-  CephContext *cct = static_cast<CephContext *>(m_local->cct());
-  m_asok_hook = new ReplayerAdminSocketHook(cct, m_peer.cluster_name, this);
```

Comment:
We would need to come up with a new naming scheme -- the uniqueness is local pool and peer, but we can't have "<pool>/<cluster>" since that conflicts with image naming format.  I am open to suggestions.